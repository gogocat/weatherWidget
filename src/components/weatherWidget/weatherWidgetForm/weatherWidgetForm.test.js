import React from 'react';
import { shallow } from 'enzyme';
import WeatherWidgetForm from './weatherWidgetForm';

const props = {
    widgetTitle: 'widget title',
    temperatureType: 'C',
    onWidgetTitleChange: jest.fn(e => {
        return 'Mocked onWidgetTitleChange';
    }),
    onTemperatureTypeChange: jest.fn(e => {
        return 'Mocked onTemperatureTypeChange function';
    }),
    onDisplayWindInfoChange: jest.fn(e => {
        return 'Mocked onDisplayWindInfoChange function';
    })
};

it('renders match snapshot', () => {
    const component = shallow(
        <WeatherWidgetForm
            widgetTitle={props.widgetTitle}
            temperatureType={props.temperatureType}
            handleTitleChange={props.onWidgetTitleChange.bind(this)}
            setTemperatureType={props.onTemperatureTypeChange.bind(this)}
            setDisplayWindInfo={props.onDisplayWindInfoChange.bind(this)}
        />
    );
    expect(component.getElements()).toMatchSnapshot();
});
