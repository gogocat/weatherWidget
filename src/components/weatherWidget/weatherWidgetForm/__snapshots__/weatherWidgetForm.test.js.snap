// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders match snapshot 1`] = `
Array [
  <div
    className="weather-widget-form"
  >
    <div
      className="weather-widget-form__row"
    >
      <label
        className="weather-widget-form__label"
        htmlFor="widgetTitle"
      >
        Title
      </label>
      <input
        className="weather-widget-form__input--text"
        id="widgetTitle"
        name="widgetTitle"
        onChange={[Function]}
        placeholder="Title of widget"
        type="text"
        value="widget title"
      />
    </div>
    <div
      className="weather-widget-form__row"
    >
      <label
        className="weather-widget-form__label"
      >
        Temperature
      </label>
      <label
        className="weather-widget-form__radio-containder"
      >
        <span
          className="weather-widget-form__radio-title"
        >
          ℃
        </span>
        <input
          checked={true}
          name="temperature"
          onChange={[Function]}
          type="radio"
          value="C"
        />
        <span
          className="weather-widget-form__radio-btn"
        />
      </label>
      <label
        className="weather-widget-form__radio-containder"
      >
        <span
          className="weather-widget-form__radio-title"
        >
          ℉
        </span>
        <input
          checked={false}
          name="temperature"
          onChange={[Function]}
          type="radio"
          value="F"
        />
        <span
          className="weather-widget-form__radio-btn"
        />
      </label>
    </div>
    <div
      className="weather-widget-form__row"
      onChange={[Function]}
    >
      <label
        className="weather-widget-form__label"
      >
        Wind
      </label>
      <label
        className="weather-widget-form__radio-containder"
      >
        <span
          className="weather-widget-form__radio-title"
        >
          On
        </span>
        <input
          defaultChecked={true}
          name="wind"
          type="radio"
          value="on"
        />
        <span
          className="weather-widget-form__radio-btn"
        />
      </label>
      <label
        className="weather-widget-form__radio-containder"
      >
        <span
          className="weather-widget-form__radio-title"
        >
          Off
        </span>
        <input
          name="wind"
          type="radio"
          value="off"
        />
        <span
          className="weather-widget-form__radio-btn"
        />
      </label>
    </div>
  </div>,
]
`;
